c_src/stb_image_nif.cpp

```diff
@@ -120,29 +115,140 @@ static ERL_NIF_TERM from_memory(ErlNifEnv *env, int argc, const ERL_NIF_TERM arg
+                ok = false;
+                break;
+            }
+        }
+
+        if (!ok) {
+            free((void *)data);
+            free((void *)frames_term);
+            free((void *)delays_term);
+            free((void *)frames_result);
+            free((void *)delays);
+            return enif_make_tuple2(env,
+                                    enif_make_atom(env, "error"),
+                                    enif_make_string(env, "out of memory", ERL_NIF_LATIN1));
+        }
+
+        ERL_NIF_TERM frames_ret = enif_make_list_from_array(env, frames_term, z);
+        ERL_NIF_TERM delays_ret = enif_make_list_from_array(env, delays_term, z);
+        ERL_NIF_TERM ret_val = enif_make_tuple4(env,
+                                                enif_make_atom(env, "ok"),
+                                                frames_ret,
+                                                enif_make_tuple3(env,
+                                                                 enif_make_int(env, y),
+                                                                 enif_make_int(env, x),
+                                                                 enif_make_int(env, 3)),
+                                                delays_ret);
+        free((void *)data);
+        free((void *)frames_term);
+        free((void *)delays_term);
+        free((void *)frames_result);
+        free((void *)delays);
+        return ret_val;
+    } else {
+        return enif_make_badarg(env);
+    }
+}
+
+static ERL_NIF_TERM to_file(ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[]) {
+    if (argc != 6) {
+        return erlang::nif::error(env, "expecting 3 arguments: filename and extension and type");
+    }
+    std::string filename, extension;
+    ErlNifBinary result;
+    int w = 0, h = 0, comp = 0;
+    if (erlang::nif::get(env, argv[0], filename) && erlang::nif::get(env, argv[1], extension) && enif_inspect_binary(env, argv[2], &result),\
+     erlang::nif::get(env, argv[3], &w), erlang::nif::get(env, argv[4], &h), erlang::nif::get(env, argv[5], &comp)) {
+        if (extension == "png") {
+            int stride_in_bytes = 0;
+            int status = stbi_write_png(filename.c_str(), w, h, comp, result.data, stride_in_bytes);
+            if (!status) {
+                return erlang::nif::error(env, "Unsuccesful attempt to write to png");
+            }
+        } else if (extension == "bmp") {
+            int status = stbi_write_bmp(filename.c_str(), w, h, comp, result.data);
+            if (!status) {
+                return erlang::nif::error(env, "Unsuccesful attempt to write to bmp");
+            }
+        } else if (extension == "tga") {
+            int status = stbi_write_tga(filename.c_str(), w, h, comp, result.data);
+            if (!status) {
+                return erlang::nif::error(env, "Unsuccesful attempt to write to tga");
+            }
+        } else if (extension == "jpg" || extension == "hdr") {
```

Comment:
I think we can just not support hdr altogether:

```suggestion
        } else if (extension == "jpg") {
```